builder/dockerfile/internals.go

```diff
@@ -202,7 +202,29 @@ func (b *Builder) runContextCommand(args []string, allowRemote bool, allowLocalD
 		}
 	}
 
-	return b.commit(container.ID, cmd, comment)
+	return b.commitWithID(container.ID, copyRunConfigWithNewCmd(b.runConfig, cmd))
+}
+
+// copyRunConfigWithNewCmd makes a copy of runConfig, sets cmd as the Cmd
+// on the copy, and then returns the copy
+func copyRunConfigWithNewCmd(runConfig *container.Config, cmd strslice.StrSlice) *container.Config {
+	copy := *runConfig
+	copy.Cmd = cmd
+	return &copy
+}
+
+func copyRunConfigWithCommentCmd(runConfig *container.Config, comment string) *container.Config {
```

Comment:
Really don' like these new functions.
Can we use func arguments instead?

ie... `copyRunconfig(WithComment(...), WithCmd(...))`